common/src/service/Fx.cs

```diff
@@ -153,10 +145,13 @@ public static Exception AssertAndFailFastService(string description)
                     // Workaround for the issue above. Throwing an unhandled exception on a separate thread to trigger process crash and crash dump collection
                     // Throwing FatalException since our service does not morph/eat up fatal exceptions
                     // We should find the tracking bug in Azure for this issue, and remove the workaround when fixed by Azure
-                    Thread failFastWorkaroundThread = new Thread(delegate ()
-                    {
-                        throw new FatalException(failFastMessage);
-                    });
+                    var failFastWorkaroundThread = new Thread(
+                        delegate ()
+                        {
+#pragma warning disable CA2219 // Do not raise exceptions in finally clauses
```

Comment:
I'm not sure wat the right fix is for this. Inside a finally you could be running code just before an exception gets thrown, so it makes sense.

I don't know what most of the Fx.cs class is for. Much of the code looks highly dubious.